tests/conftest.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+import inspect
+import socket
+from collections import namedtuple
+
+from cqc.pythonLib import CQCConnection
+
+Call = namedtuple("Call", ["name", "args", "kwargs"])
+
+
+def _spy_wrapper(method):
+    """Wraps a method to be able to spy on it"""
+    def new_method(self, *args, **kwargs):
+        if method.__name__ == '__init__':
+            self.calls = []
+        call = Call(method.__name__, args, kwargs)
+        self.calls.append(call)
+        return method(self, *args, **kwargs)
+
+    return new_method
+
+
+def spy_on_class(cls):
+    """Spies on all calls to the methods of a class"""
+    for method_name, method in inspect.getmembers(cls, predicate=inspect.isfunction):
+        setattr(cls, method_name, _spy_wrapper(method))
+    return cls
+
+
+@spy_on_class
+class MockSocket:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def connect(self, *args, **kwargs):
+        pass
+
+    def send(self, *args, **kwargs):
+        pass
+
+    def recv(self, *args, **kwargs):
+        pass
+
+    def close(self, *args, **kwargs):
+        pass
+
+
+@pytest.fixture
+def mock_socket(monkeypatch):
+    def get_mocked_socket(*args, **kwargs):
+        mock_socket = MockSocket(*args, **kwargs)
+        return mock_socket
+
+    monkeypatch.setattr(socket, "socket", get_mocked_socket)
+
+
+class MockedFirstMessage:
+    """Mocks the second header returned by CQCConnection.readMessage"""
```

Comment:
Is this supposed to be **first** header?

Reply:
Done